parking/backend/db/sql_constants.py

```diff
@@ -63,6 +80,50 @@
  AND num_allocated < num_available
 """
 
+PARKINGLOTS_DECREMENT_ALLOCATION = """
+UPDATE ParkingLots
+SET num_allocated = num_allocated - 1
+WHERE id = $1
+"""
+
+PARKINGLOTS_SELECT = """
+SELECT * FROM ParkingLots
+WHERE id = $1
+"""
+
+PARKINGLOTS_SELECT_WITHIN_DISTANCE = """
+SELECT *
+FROM (
+    SELECT
+        *, earth_distance(ll_to_earth($1, $2), ll_to_earth(lat, long)) as distance
+    FROM
+        ParkingLots
+    WHERE
+        earth_box(ll_to_earth($1, $2), $3) @> ll_to_earth(lat, long) 
+        AND num_available > num_allocated
```

Comment:
num_available is a bit of a misleading name, since num_allocated of them are in fact not available. Should be something like capacity, with the order of the equality reversed:
`num_allocated < capacity`
This is a lot more readable in my opinion

Reply:
In this system:
capacity = total parking spaces in a lot
num_available = number of parking spaces available as per sensors
num_allocated = number of parking spaces allocated by our system

We need all of these to make decisions and `num_allocated < capacity` wouldn't work.

Reply:
Okay, that makes more sense